File: apps/flutter/lib/core/messages/caregiver_message.dart
class CaregiverMessage {
  final String id;

  /// NEW: links a message to a patient
  final String patientId;

  final String sender;
  final String subject;
  final String preview;
  final DateTime sentAt;
  final bool unread;

  const CaregiverMessage({
    required this.id,
    required this.patientId,
    required this.sender,
    required this.subject,
    required this.preview,
    required this.sentAt,
    required this.unread,
  });
}

File: apps/react-native/src/models/CaregiverMessage.js
/**
 * Message Model
 * Equivalent to Flutter's CaregiverMessage class
 */

export class CaregiverMessage {
  constructor(id, sender, subject, preview, sentAt, unread = false) {
    this.id = id;
    this.sender = sender;
    this.subject = subject;
    this.preview = preview;
    this.sentAt = sentAt;
    this.unread = unread;
  }
}

export default CaregiverMessage;
